virtualmachine/vsphere/vm.go

```
@@ -837,56 +836,163 @@ func DeleteTemplate(vm *VM) error {
+		})
+	}
+	return datastoreList, nil
+}
 
-		// get the network names
-		var network mo.Network
-		for _, networkMor := range networksMor {
-			switch networkMor.Type {
-			case "Network":
-				err = vm.collector.RetrieveOne(vm.ctx, networkMor, []string{"name"}, &network)
-				if err != nil {
-					return nil, err
-				}
-				networkList[dc.Name()] = append(networkList[dc.Name()], network.Name)
+// GetNetworkInHost : Returns the networks in a host in a cluster
+func GetNetworkInHost(vm *VM) ([]map[string]string, error) {
+	var (
+		networkList []map[string]string
+		networkMap  map[string]string
+		network     mo.Network
+		portGroup   mo.DistributedVirtualPortgroup
+		opNetwork   mo.OpaqueNetwork
+		hsMo        mo.HostSystem
+	)
+	// set up session to vcenter server
+	if err := SetupSession(vm); err != nil {
+		return nil, err
+	}
+
+	dc, err := GetDatacenter(vm)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the cluster resource and its host, network and datastore
+	crMo, err := findClusterComputeResource(vm, dc, vm.Destination.DestinationName)
+	if err != nil {
+		return nil, err
+	}
+
+	// find the host in Destination.HostSystem
+	for _, host := range crMo.Host {
+		err = vm.collector.RetrieveOne(vm.ctx, host, []string{"name", "network"}, &hsMo)
+		if err != nil {
+			return nil, err
+		}
+		if hsMo.Name == vm.Destination.HostSystem {
+			break
+		}
+	}
+
+	// Add all the networks in host to network list
+	for _, networkMor := range hsMo.Network {
+		switch networkMor.Type {
+		case "Network":
+			err = vm.collector.RetrieveOne(vm.ctx, networkMor, []string{"name"}, &network)
+			if err != nil {
+				return nil, err
+			}
+			networkMap = map[string]string{"name": network.Name}
+		case "DistributedVirtualPortgroup":
+			err = vm.collector.RetrieveOne(vm.ctx, networkMor, []string{"name"}, &portGroup)
+			if err != nil {
+				return nil, err
+			}
+			networkMap = map[string]string{"name": portGroup.Name}
+		case "OpaqueNetwork":
```

Can we have a default case handling?
At least print the network type, rather than silently ignoring.